Mathlib/Logic/Equiv/Fin.lean

```diff
@@ -498,3 +498,12 @@ instance subsingleton_fin_zero : Subsingleton (Fin 0) :=
 /-- `Fin 1` is a subsingleton. -/
 instance subsingleton_fin_one : Subsingleton (Fin 1) :=
   finOneEquiv.subsingleton
+
+/-- The natural `Equiv` between `(Fin m → α) × (Fin n → α)` and `(Fin (m + n) → α)`.-/
+@[simps]
+def Equiv.finArrowProdEquivFinAddArrow {α : Type*} (m n : ℕ) :
```

Comment:
I suspect a better name for this would be `Fin.appendEquiv`. Perhaps @YaelDillies has opinions on naming here.

Reply:
Yes I agree. That would match `Fin.snocEquiv` and friends. Do you think you could also make it dependent?

Reply:
`Fin.append` is not dependent so I think that is out of scope.

Reply:
Right